pyodata/v2/service.py

```diff
@@ -874,16 +874,10 @@ def nav(self, nav_property):
             association_info.name,
             association_info.namespace)
 
-        navigation_entity_set = None
-        for end in association_set.end_roles:
-            if association_set.end_by_entity_set(end.entity_set_name).role == navigation_property.to_role.role:
-                navigation_entity_set = self._service.schema.entity_set(end.entity_set_name, association_info.namespace)
+        end = association_set.end_by_role(navigation_property.to_role.role)
+        navigation_entity_set = self._service.schema.entity_set(end.entity_set_name)
 
-        if not navigation_entity_set:
-            raise PyODataException(f'No association set for role {navigation_property.to_role}')
```

Comment:
If the bug is about "many to one relationships", why remove this exception rise? Even the second part of the PR description - "it is not possible to use the nav method with associations where both ends are of the same entity type." - as I understand it, the case that you would like to refer to something that does not exist at all is still valid. 

As an example - consider metadata file, that is invalid, but [forced to be parsed anyway ](https://github.com/SAP/python-pyodata/blob/master/docs/usage/initialization.rst#dealing-with-errors-during-parsing-metadata)  - and then you can end easily with trying to access something which is not there at all.

It bothers me a bit that there is not a test that would start to fail by removing these lines - you are only adding new tests after all.

Reply:
I have removed this exception because the end_by_role method is [raising a KeyError](https://github.com/SAP/python-pyodata/blob/9a72f930800ed758cf8a4d9c53a8c2434ff49133/pyodata/v2/model.py#L2165) in case the end role was not found. I have thought about to handle the KeyError and raising the PyODataException, but this would be a bit redundant. However, due to the possible KeyError of end_by_role the navigation_entity_set can never be None (e.g. in case of an invalid metadata file) and the PyODataException would never be raised.